tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -716,6 +716,53 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system",
 	},
 
+	/** Firmware layout for the A7 v5*/
+	{
+		.id = "ARCHER-A7-V5",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:Archer A7,product_ver:5.0.0,special_id:45550000}\n"
+			"{product_name:Archer A7,product_ver:5.0.0,special_id:55530000}\n"
+			"{product_name:Archer A7,product_ver:5.0.0,special_id:43410000}\n"
+			"{product_name:Archer A7,product_ver:5.0.0,special_id:4A500000}\n"
+			"{product_name:Archer A7,product_ver:5.0.0,special_id:54570000}\n",
+
+		.support_trail = '\x00',
+		.soft_ver = "soft_ver:1.0.0\n",
+
+		/**
+		  We use a bigger os-image partition than the stock images (and thus
+		  smaller file-system), as our kernel doesn't fit in the stock firmware's
+		  1MB os-image.
+		  */
+		.partitions = {
+			{"factory-boot",    0x00000,  0x20000},
+			{"fs-uboot",        0x20000,  0x20000},
+
+			{"os-image",        0x40000,  0x180000}, /* Stock: base 0x40000  size 0x120000 */
+			{"file-system",     0x1c0000, 0xd80000}, /* Stock: base 0x160000 size 0xde0000 */
```

Comment:
Could dynamic partitioning be used here by deleting the os-image and file-system partitions and replacing them with `{"firmware", 0x40000, 0xf00000},`? It looks like build_image() can create the two partitions on the fly.

Reply:
you could, but i'd rather copy paste from the other devices. It's already difficult enough to get something upstream. Probably easier when there's nothing new.

Reply:
+1 for adding dynamic partitioning.

I'm currently in the process of adapting all ar71xx targets to use this, as the os-image partition size of 0x180000 is too small and causes build errors using kernel 4.14

Reply:
@aep 

I pushed the changes for the other ar71xx targets, switching most of them to dynamic partitioning.

Reply:
Thank you for your comment! I've been putzing about, unable to create the *factory.bin.

I have just one quick question. To which patch are you trying to link? Chrome shows "**this patch**" in bold and elinks displays it as just grey text.

Also, thank you to aep!

Reply:
To be honest, I had a good bit of help from the IRC channel.  They pointed me at [this patch](https://patchwork.ozlabs.org/patch/995935/).  In fact, I believe they crated a patch link for me.  I think I recall them saying it was from the mailing list as opposed to the pull request.

Using the patch linked here, I was able to follow my recipe pasted above.  Good luck!

Reply:
If I want to be able to `opkg install` stuff, will this version work? 

I assume `ath79` device tree hasn't been released, so we won't have opkg repository for it, is that right?

Reply:
I've been installing downloaded .ipk files.